feat: Add option to check any ERC20 token balance

Pratham-19 opened the pull request:
Implementing 
- Check any token balance 
- Check balance for any address


1. Check balance 

- Check any ERC20 token balance by default using the stored balance else can also provide address using
`rsk-cli balance --contract <address> --address (optional) -t`

![image](https://github.com/user-attachments/assets/9ec8ec13-c632-43a2-a311-d26f221c7b20)

- Added contract validation 
![image](https://github.com/user-attachments/assets/ecc81cfb-eefd-4edb-8216-3be0e5e68bba)

chrisarevalodev commented:
Hey @Pratham-19! 

I’m really impressed with your improvements to the CLI—they look fantastic. I have a few comments about your implementation that are critical to address before we can merge your PR.

I’ve already tested it, and it works well! However, there are some aspects of the implementation that don’t align with the existing CLI structure and the plans outlined for upcoming features in our backlog.

Specifically, you created a new program called `token` in the `bin/index.ts` file, where you can call both balance and transfer commands for custom tokens. While this is a great addition, there are already existing commands named `balance` and `transfer`. The goal is to update these commands so that users can execute them as follows:

- Balance Command:
Run `rsk-cli balance` or `rsk-cli balance --testnet`. Using `inquirer` and `ora`, the CLI should prompt the user to choose between rBTC, RIF, USD RIF, DoC, or a custom token. If the user selects “custom”, the CLI should then ask for the contract address using `inquirer`.

- Transfer Command:
Similarly, run `rsk-cli transfer` or `rsk-cli transfer --testnet`. Here too, the CLI should prompt the user to specify the token type (rBTC, RIF, etc.) or provide a contract address for custom tokens and also the amount to be transferred.

For reference, there is a very similar implementation using `inquirer` and `ora` in the contract command, which you may find helpful as a guide.

Thanks again for your hard work! You’re doing an amazing job. If you have any questions or need further clarification, feel free to reach out.

Pratham-19 commented:
Thanks for your comment @chrisarevalodev 

I got the requirements, I'll revamp my code and add inquirer and ora for the same

Pratham-19 commented:
Hey @chrisarevalodev ,

These are the updates

- Added support for rBTC, RIF, USD RIF, DoC, and custom tokens on both mainnet and testnet (need help regarding testnet contract addressfor these tokens)


![image](https://github.com/user-attachments/assets/6ad2169c-33fb-478d-99cb-311f95c39b38)


![image](https://github.com/user-attachments/assets/2ba0bb29-7ab4-42c1-a39c-972e79aef0cc)

- Added support for checking balance of any specified address

![image](https://github.com/user-attachments/assets/2543c3ea-c5e6-4908-8f1d-f24f38b0952e)

- Added contract address validation

![image](https://github.com/user-attachments/assets/ee666bad-48ad-4f25-a86f-6847c5e4236f)

chrisarevalodev commented:
Hey @Pratham-19, I'm leaving my comments here as a thread

Pratham-19 commented:
Hey @chrisarevalodev ,

I have added a function to check whether the contract is erc20, so it will only work for all the ERC20 contract

![image](https://github.com/user-attachments/assets/d6ad7377-a58e-4985-82e9-91bb5d48c901)

chrisarevalodev reviewed (approved):
Tested and working. LGTM @ezequiel-rodriguez

ezequiel-rodriguez reviewed (approved):
LGTM
